src/phases/query/CollScanPredicateSelectivity.yml

```diff
@@ -0,0 +1,688 @@
+        quarterPositiveInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [25, 75]}},
+        halfPositiveInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [50, 50]}},
+        quarterNegativeInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [75, 25]}},
+        tenPercentNegativeInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [90, 10]}},
+        onePercentNegativeInt: {^Choose: {from: *PositiveAndNegativeIntegers, weights: [99, 1]}},
+      }
+    ]
+  }
+
+ActorTemplates:
+- TemplateName: FindQueryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "unused"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: *Repeat
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {^Parameter: {Name: "Filter", Default: "undefinedFilter"}}
+
+Actors:
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document: *Document
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Heuristic CE defines the following selectivity order for predicates:
+# $eq < closed range (e.g. {$gt: 10, $lte: 20}) < open range (e.g. {$gt: 10}) < $ne
+
+- ActorFromTemplate:
+    TemplateName: FindQueryTemplate
+    TemplateParameters:
+      Name: HighSelHeuristicGoodTwoRangeClauses
+      ActivePhase: 3
+      Filter:
+        # Here (0, 1e9) is the most selective predicate, both in reality and according to heuristic CE.
+        &HighSelHeuristicGoodTwoRangeClauses {
+          onePercentPositiveInt: {$gt: 0, $lt: *Billion}, # Selectivity = .01
+          onePercentNegativeInt: {$gt: 0}, # Selectivity = .99
+        }
+
+- ActorFromTemplate:
+    TemplateName: FindQueryTemplate
+    TemplateParameters:
+      Name: HighSelHeuristicGoodTwoEqClauses
+      ActivePhase: 4
+      Filter:
+        &HighSelHeuristicGoodTwoEqClauses {
+          onePercentNull: null, # Selectivity = .01
+          onePercentStr: {$ne: "non-null"}, # Selectivity = .99
+        }
```

Comment:
@alyacb I added another version of each two-predicate query. The reordering PR will probably reorder the $eq/$ne queries, but as we discussed briefly offline, it might not reorder e.g. `HighSelHeuristicGoodTwoRangeClauses` because `{$gt: 0, $lt: *Billion}` is seen as a conjunction of open-range intervals instead of one closed-range interval.

Reply:
Good point, but I have a follow-up question- do we know if SBE/classic use the original order in any of these cases?